packages/flutter/lib/src/widgets/image.dart

```diff
@@ -1119,22 +1120,29 @@ class _ImageState extends State<Image> with WidgetsBindingObserver {
     _updateSourceStream(newStream);
   }
 
-  ImageStreamListener _getListener([ImageLoadingBuilder loadingBuilder]) {
-    loadingBuilder ??= widget.loadingBuilder;
-    _lastException = null;
-    _lastStack = null;
-    return ImageStreamListener(
-      _handleImageFrame,
-      onChunk: loadingBuilder == null ? null : _handleImageChunk,
-      onError: widget.errorBuilder != null
-        ? (dynamic error, StackTrace stackTrace) {
-            setState(() {
-              _lastException = error;
-              _lastStack = stackTrace;
-            });
-          }
-        : null,
-    );
+  ImageStreamListener _imageStreamListener;
+  ImageStreamListener _getListener() {
```

Comment:
I believe we still need to retain the old widget's loading builder here.

Reply:
It used to get the old widget's loading builder, because it tries to replicate the old ImageStreamListener with that. Now that ImageStreamListener is stored you don't need to pass that loading builder anymore, because it is retained in _imageStreamListener.